files/build_templates/docker_image_ctl.j2

```diff
@@ -169,18 +177,63 @@ start() {
 
     {%- if docker_container_name == "database" %}
     echo "Creating new {{docker_container_name}}$DEV container"
+    if [ -z "$DEV" ]; then
+        # if database_global exists in old_config, use it; otherwise use the default one in new image
+        if [ -f /etc/sonic/old_config/database_global.json ]; then
+            echo "Use database_global.json from old system..."
+            mv /etc/sonic/old_config/database_global.json /etc/sonic/
+        fi
+    fi
     # if database_config exists in old_config, use it; otherwise use the default one in new image
-    if [ -f /etc/sonic/old_config/database_config.json ]; then
+    if [ -f /etc/sonic/old_config/database_config$DEV.json ]; then
         echo "Use database_config.json from old system..."
-        mv /etc/sonic/old_config/database_config.json /etc/sonic/
+        mv /etc/sonic/old_config/database_config$DEV.json /etc/sonic/
     fi
     {%- else %}
     echo "Creating new {{docker_container_name}}$DEV container with HWSKU $HWSKU"
     {%- endif %}
 
+    # In Multi ASIC platforms the global database config file database_global.json will exist.
+    # Parse the file and get the include path for the database_config.json files used in
+    # various namesapces. The database_config paths are relative to the DIR of SONIC_DB_GLOBAL_JSON.
+    SONIC_DB_GLOBAL_JSON="/var/run/redis/sonic-db/database_global.json"
+    if [ -f "$SONIC_DB_GLOBAL_JSON" ]; then
+        redis_dir_list=`/usr/bin/python -c "import sys; import os; import json; f=open(sys.argv[1]); \
```

Comment:
suggest to have a python script to do this , then at this point, we just need to call the script which will make it easy to read/debug/test and make changes in future, don't need to build the whole system. The script could be in swsssdk library the same level as sonic-db-cli

Reply:
Sure, I can take this up as a future task to move to a python script. Hope it is fine.

Reply:
Please add TODO comment or Github issue for future improvement.

Reply:
I have added a TODO comment to have a python script to do this logic instead of using python -c